Python fails to build on macOS 10.13.6, with MacOSX10.14.sdk command line tools.

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [ ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Install python with `brew install python`, on macOS 10.13.6, with MacOSX10.14.sdk command line tools.
- What happened (include command output)
Gist-logs: <https://gist.github.com/138b058432ac0e7ef3cc01af128389ac>
Build error for python:
```
==> make
==> make install PYTHONAPPSDIR=/Users/ling/homebrew/Cellar/python/3.7.0
Last 15 lines from /Users/ling/Library/Logs/Homebrew/python/03.make:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSFileManager.h:472:147: error: expected ';' after top level declarator
FOUNDATION_EXPORT NSFileProtectionType const NSFileProtectionCompleteUntilFirstUserAuthentication API_AVAILABLE(ios(5.0), watchos(2.0), tvos(9.0)) API_PROHIBITED(macos);
                                                                                                                                                  ^
In file included from ./FileSettings.m:9:
In file included from ./FileSettings.h:9:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:39:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSHashTable.h:5:
/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSPointerFunctions.h:36:100: error: missing ',' between enumerators
    NSPointerFunctionsZeroingWeakMemory API_DEPRECATED("GC no longer supported", macos(10.5, 10.8)) API_PROHIBITED(ios, watchos, tvos) = (1UL << 0),  // deprecated; uses GC weak read and write barriers, and dangling pointer behavior otherwise
                                                                                                   ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [FileSettings.o] Error 1
make[1]: *** [install_PythonLauncher] Error 2
make: *** [frameworkinstallapps] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
pipenv cannot easily manage multiple python versions https://github.com/Homebrew/homebrew-core/issues/31487
WeeChat fails to build --with-python https://github.com/Homebrew/homebrew-core/issues/30509
Jupyter notebook is launched with PYTHONPATH, modifies sys.path in Python kernels https://github.com/Homebrew/homebrew-core/issues/31325
`vim` fails to build from source with option `--with-client-server` when python is installed with '--with-tcl-tk' https://github.com/Homebrew/homebrew-core/issues/30949
```
- What you expected to happen
See if someone has same problem.
- Step-by-step reproduction instructions (by running `brew install` commands)
On macOS 10.13.6, with MacOSX10.14.sdk command line tools, just run `brew install python`

FWIW, I managed to build `python` with Xcode 10 by removing the following line from `python.rb`
https://github.com/Homebrew/homebrew-core/blob/9a7270788f73e532a4841f3f0b5dc2642a9eaccf/Formula/python.rb#L107

It seems that with Xcode 10, `-isysroot` and `-I/usr/include` don't seem to go well together.

```console
$ cat x.m 
#include <Foundation/Foundation.h>
$ clang -c x.m # OK
$ clang -I/usr/include -c x.m # OK
$ clang -isysroot /Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -c x.m # OK
$ clang -isysroot /Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -I/usr/include -c x.m # KO
In file included from x.m:1:
In file included from /Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:87:
In file included from /Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:17:
In file included from /Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:29:
/Applications/Xcode-10.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AppleEvents.h:597:165: error: 
      expected function body after function declarator
  ...AEEventID theAEEventID, Boolean askUserIfNeeded ) __OSX_AVAILABLE_STARTI...
```

I have a similar problem. I can't get python3.7 to install with tcl-tk on macOS 10.13.6

$ brew install python3 --with-tcl-tk
......

/usr/bin/install -c -s ../python.exe "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/MacOS/Python"
sed -e "s!%bundleid%!org.python.python!g" \
		-e "s!%version%!`DYLD_FRAMEWORK_PATH=/private/tmp/python-20180920-93810-rk2qwr/Python-3.7.0 ../python.exe \
		-c 'import platform; print(platform.python_version())'`!g" \
		< "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/Info.plist.in" \
		> "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/Info.plist"
rm "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/Info.plist.in"
cd PythonLauncher && make install DESTDIR=
sed 's/%VERSION%/'"`DYLD_FRAMEWORK_PATH=/private/tmp/python-20180920-93810-rk2qwr/Python-3.7.0 ../../python.exe -c 'import platform; print(platform.python_version())'`"'/g' < ./Info.plist.in > Info.plist
clang -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -I/usr/include -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -o FileSettings.o -c ./FileSettings.m
In file included from ./FileSettings.m:9:
In file included from ./FileSettings.h:9:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:87:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSURLError.h:17:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/CoreServices.framework/Headers/CoreServices.h:24:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AE.h:29:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/CoreServices.framework/Frameworks/AE.framework/Headers/AppleEvents.h:597:165: error: expected function body after function declarator
extern OSStatus AEDeterminePermissionToAutomateTarget( const AEAddressDesc* target, AEEventClass theAEEventClass, AEEventID theAEEventID, Boolean askUserIfNeeded ) __OSX_AVAILABLE_STARTING( __MAC_10_14, __IPHONE_NA );
                                                                                                                                                                    ^
/usr/include/Availability.h:216:50: note: expanded from macro '__OSX_AVAILABLE_STARTING'
    #define __OSX_AVAILABLE_STARTING(_osx, _ios) __AVAILABILITY_INTERNAL##_osx
                                                 ^
<scratch space>:58:1: note: expanded from here
__AVAILABILITY_INTERNAL__MAC_10_14
^
In file included from ./FileSettings.m:9:
In file included from ./FileSettings.h:9:
In file included from /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:179:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:21:27: error: expected a version of the form 'major[.minor[.subminor]]'
} NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API");
                          ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:23:25: error: expected a version of the form 'major[.minor[.subminor]]'
NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API")
                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:102:26: error: expected a version of the form 'major[.minor[.subminor]]'
NS_DEPRECATED_MAC(10_10, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API")
                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:115:95: error: expected a version of the form 'major[.minor[.subminor]]'
FOUNDATION_EXPORT NSString * const NSUserNotificationDefaultSoundName NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API") API_UNAVAILABLE(ios, watchos, tvos);
                                                                                              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:117:25: error: expected a version of the form 'major[.minor[.subminor]]'
NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API")
                        ^
6 errors generated.
make[2]: *** [FileSettings.o] Error 1
make[1]: *** [install_PythonLauncher] Error 2
make: *** [frameworkinstallapps] Error 2
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `write': Broken pipe (Errno::EPIPE)
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `puts'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `rescue in block (3 levels) in safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:31:in `block (3 levels) in safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:30:in `fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:30:in `block (2 levels) in safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:27:in `open'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:27:in `block in safe_fork'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/tmpdir.rb:89:in `mktmpdir'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:26:in `safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:719:in `build'
	from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:311:in `install'
	from /usr/local/Homebrew/Library/Homebrew/reinstall.rb:33:in `reinstall_formula'
	from /usr/local/Homebrew/Library/Homebrew/cmd/reinstall.rb:26:in `block in reinstall'
	from /usr/local/Homebrew/Library/Homebrew/cmd/reinstall.rb:20:in `each'
	from /usr/local/Homebrew/Library/Homebrew/cmd/reinstall.rb:20:in `reinstall'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'

==> Formula
Path: /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/python.rb
==> Configuration
HOMEBREW_VERSION: 1.7.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: d639559c4de48faad79d3ecb1ec84dc851ea66f2
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 68763193b0a0973391c814b1b11720ef72fadc33
Core tap last commit: 7 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_GIT: git
CPU: quad-core 64-bit arrandale
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.19.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 10.0.2
macOS: 10.13.6-x86_64
CLT: 10.0.0.0.1.1535735448
CLT headers: N/A
Xcode: 10.0
XQuartz: 2.7.11 => /opt/X11
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
CMAKE_PREFIX_PATH: /usr/local/opt/sphinx-doc:/usr/local/opt/openssl:/usr/local/opt/readline:/usr/local/opt/sqlite:/usr/local/opt/tcl-tk:/usr/local
CMAKE_INCLUDE_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/libxml2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_PATH: /usr/local/opt/openssl/lib/pkgconfig:/usr/local/opt/sqlite/lib/pkgconfig:/usr/local/opt/xz/lib/pkgconfig:/usr/local/opt/tcl-tk/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.13
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/pkg-config/bin:/usr/local/opt/sphinx-doc/bin:/usr/local/opt/gdbm/bin:/usr/local/opt/openssl/bin:/usr/local/opt/sqlite/bin:/usr/local/opt/xz/bin:/usr/local/opt/tcl-tk/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: python 3.7.0 did not build
Logs:
     /Users/belinda/Library/Logs/Homebrew/python/00.options.out
     /Users/belinda/Library/Logs/Homebrew/python/03.make.cc
     /Users/belinda/Library/Logs/Homebrew/python/01.configure.cc
     /Users/belinda/Library/Logs/Homebrew/python/02.make.cc
     /Users/belinda/Library/Logs/Homebrew/python/config.log
     /Users/belinda/Library/Logs/Homebrew/python/01.configure
     /Users/belinda/Library/Logs/Homebrew/python/03.make
     /Users/belinda/Library/Logs/Homebrew/python/02.make
READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
pipenv cannot easily manage multiple python versions https://github.com/Homebrew/homebrew-core/issues/31487
WeeChat fails to build --with-python https://github.com/Homebrew/homebrew-core/issues/30509
Python fails to build on macOS 10.13.6, with MacOSX10.14.sdk command line tools. https://github.com/Homebrew/homebrew-core/issues/31679
Jupyter notebook is launched with PYTHONPATH, modifies sys.path in Python kernels https://github.com/Homebrew/homebrew-core/issues/31325
`vim` fails to build from source with option `--with-client-server` when python is installed with '--with-tcl-tk' https://github.com/Homebrew/homebrew-core/issues/30949

Having the same issue.
```
Last 15 lines from /Users/maxwell/Library/Logs/Homebrew/python/03.make:
NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API")
                        ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:102:26: error: expected a version of the form 'major[.minor[.subminor]]'
NS_DEPRECATED_MAC(10_10, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API")
                         ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:115:95: error: expected a version of the form 'major[.minor[.subminor]]'
FOUNDATION_EXPORT NSString * const NSUserNotificationDefaultSoundName NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API") API_UNAVAILABLE(ios, watchos, tvos);
                                                                                              ^
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSUserNotification.h:117:25: error: expected a version of the form 'major[.minor[.subminor]]'
NS_DEPRECATED_MAC(10_8, API_TO_BE_DEPRECATED, "All NSUserNotifications API should be replaced with UserNotifications.frameworks API")
                        ^
6 errors generated.
make[2]: *** [FileSettings.o] Error 1
make[1]: *** [install_PythonLauncher] Error 2
make: *** [frameworkinstallapps] Error 2
/usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `write': Broken pipe (Errno::EPIPE)
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `puts'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:49:in `rescue in block (3 levels) in safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:31:in `block (3 levels) in safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:30:in `fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:30:in `block (2 levels) in safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:27:in `open'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:27:in `block in safe_fork'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/tmpdir.rb:89:in `mktmpdir'
	from /usr/local/Homebrew/Library/Homebrew/utils/fork.rb:26:in `safe_fork'
	from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:719:in `build'
	from /usr/local/Homebrew/Library/Homebrew/formula_installer.rb:311:in `install'
	from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:321:in `install_formula'
	from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:253:in `block in install'
	from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:251:in `each'
	from /usr/local/Homebrew/Library/Homebrew/cmd/install.rb:251:in `install'
	from /usr/local/Homebrew/Library/Homebrew/brew.rb:89:in `<main>'
```

--- a/Formula/python@2.rb
+++ b/Formula/python@2.rb
@@ -111,7 +111,9 @@ class PythonAT2 < Formula
       # The setup.py looks at "-isysroot" to get the sysroot (and not at --sysroot)
       cflags  << "-isysroot #{MacOS.sdk_path}"
       ldflags << "-isysroot #{MacOS.sdk_path}"
-      cflags  << "-I/usr/include" # find zlib
+      if DevelopmentTools.clang_build_version < 1000
+        cflags  << "-I/usr/include" # find zlib
+      end
       # For the Xlib.h, Python needs this header dir with the system Tk
       if build.without? "tcl-tk"
         # Yep, this needs the absolute path where zlib needed

